diff --git a/sdk/bpf/c/bpf.mk b/sdk/bpf/c/bpf.mk
index 541629ad49..5046be7884 100644
--- a/sdk/bpf/c/bpf.mk
+++ b/sdk/bpf/c/bpf.mk
@@ -14,6 +14,12 @@ TEST_PREFIX ?= test_
 OUT_DIR ?= ./out
 OS := $(shell uname)
 
+ifeq ($(V),1)
+TEST_FLAGS ?= --verbose
+else
+TEST_FLAGS ?=
+endif
+
 LLVM_DIR = $(LOCAL_PATH)../dependencies/bpf-tools/llvm
 LLVM_SYSTEM_INC_DIRS := $(LLVM_DIR)/lib/clang/13.0.0/include
 COMPILER_RT_DIR = $(LOCAL_PATH)../dependencies/bpf-tools/rust/lib/rustlib/bpfel-unknown-unknown/lib
@@ -33,6 +39,9 @@ SYSTEM_INC_DIRS := \
   $(LLVM_SYSTEM_INC_DIRS) \
 
 C_FLAGS := \
+  -Wall \
+  -Wextra \
+  -Wconversion \
   -Werror \
   -O2 \
   -fno-builtin \
@@ -40,6 +49,8 @@ C_FLAGS := \
   $(addprefix -isystem,$(SYSTEM_INC_DIRS)) \
   $(addprefix -I,$(STD_INC_DIRS)) \
   $(addprefix -I,$(INC_DIRS)) \
+  -DSOLANA=1 \
+  -D__GLIBC_USE\(F\)=__GLIBC_USE_\#\#F \
 
 ifeq ($(SOL_SBFV2),1)
 C_FLAGS := \
@@ -68,6 +79,7 @@ BPF_CXX_FLAGS := \
   -march=bpfel+solana
 
 BPF_LLD_FLAGS := \
+  -z defs \
   -z notext \
   -shared \
   --Bdynamic \
@@ -245,6 +257,7 @@ define TEST_EXEC_RULE
 $1: $2
 	LD_LIBRARY_PATH=$(TESTFRAMEWORK_RPATH) \
 	$2$(\n)
+	$2 $(TEST_FLAGS)$(\n)
 endef
 
 .PHONY: $(INSTALL_SH)
