diff --git a/sdk/bpf/c/bpf.mk b/sdk/bpf/c/bpf.mk
index 4b3039db1..018e8deb4 100644
--- a/sdk/bpf/c/bpf.mk
+++ b/sdk/bpf/c/bpf.mk
@@ -14,6 +14,12 @@ TEST_PREFIX ?= test_
 OUT_DIR ?= ./out
 OS := $(shell uname)
 
+ifeq ($(V),1)
+TEST_FLAGS ?= --verbose
+else
+TEST_FLAGS ?=
+endif
+
 LLVM_DIR = $(LOCAL_PATH)../dependencies/bpf-tools/llvm
 LLVM_SYSTEM_INC_DIRS := $(LLVM_DIR)/lib/clang/12.0.1/include
 COMPILER_RT_DIR = $(LOCAL_PATH)../dependencies/bpf-tools/rust/lib/rustlib/bpfel-unknown-unknown/lib
@@ -31,6 +37,9 @@ SYSTEM_INC_DIRS := \
   $(LLVM_SYSTEM_INC_DIRS) \
 
 C_FLAGS := \
+  -Wall \
+  -Wextra \
+  -Wconversion \
   -Werror \
   -O2 \
   -fno-builtin \
@@ -59,6 +68,7 @@ BPF_CXX_FLAGS := \
   -march=bpfel+solana
 
 BPF_LLD_FLAGS := \
+  -z defs \
   -z notext \
   -shared \
   --Bdynamic \
@@ -195,7 +205,7 @@ endef
 define TEST_EXEC_RULE
 $1: $2
 	LD_LIBRARY_PATH=$(TESTFRAMEWORK_RPATH) \
-	$2$(\n)
+	$2 $(TEST_FLAGS)$(\n)
 endef
 
 .PHONY: $(INSTALL_SH)
diff --git a/sdk/bpf/c/inc/solana_sdk.h b/sdk/bpf/c/inc/solana_sdk.h
index b5cad9833..b3b496123 100644
--- a/sdk/bpf/c/inc/solana_sdk.h
+++ b/sdk/bpf/c/inc/solana_sdk.h
@@ -565,6 +565,7 @@ uint64_t sol_try_find_program_address(
 /**
  * Internal cross-program invocation function
  */
+__attribute__(( weak ))
 uint64_t sol_invoke_signed_c(
   const SolInstruction *instruction,
   const SolAccountInfo *account_infos,
